tests/components/nut/test_sensor.py

```diff
@@ -17,210 +19,36 @@
-
-
-async def test_dl650elcd(hass: HomeAssistant) -> None:
-    """Test creation of DL650ELCD sensors."""
-
-    config_entry = await async_init_integration(hass, "DL650ELCD")
-    registry = er.async_get(hass)
-    entry = registry.async_get("sensor.ups1_battery_charge")
-    assert entry
-    assert entry.unique_id == f"{config_entry.entry_id}_battery.charge"
-
-    state = hass.states.get("sensor.ups1_battery_charge")
-    assert state.state == "100"
-
-    expected_attributes = {
-        "device_class": "battery",
-        "friendly_name": "Ups1 Battery charge",
-        "unit_of_measurement": PERCENTAGE,
-    }
-    # Only test for a subset of attributes in case
-    # HA changes the implementation and a new one appears
-    assert all(
-        state.attributes[key] == attr for key, attr in expected_attributes.items()
-    )
-
-
-async def test_eaton5p1550(hass: HomeAssistant) -> None:
-    """Test creation of EATON5P1550 sensors."""
+@pytest.mark.parametrize(
+    ("model", "unique_id"),
+    [
+        ("PR3000RT2U", "CPS_PR3000RT2U_PYVJO2000034_battery.charge"),
+        ("CP1350C", ""),
+        ("5E850I", ""),
+        ("5E650I", ""),
+        (
+            "BACKUPSES600M1",
+            "American Power Conversion_Back-UPS ES 600M1_4B1713P32195 _battery.charge",
+        ),
+        ("CP1500PFCLCD", ""),
+        ("DL650ELCD", ""),
+        ("EATON5P1550", ""),
+        ("blazer_usb", ""),
+    ],
+)
+async def test_devices(
+    hass: HomeAssistant, entity_registry: er.EntityRegistry, model: str, unique_id: str
+) -> None:
+    """Test creation of device sensors."""
 
-    config_entry = await async_init_integration(hass, "EATON5P1550")
-    registry = er.async_get(hass)
-    entry = registry.async_get("sensor.ups1_battery_charge")
+    config_entry = await async_init_integration(hass, model)
+    entry = entity_registry.async_get("sensor.ups1_battery_charge")
     assert entry
-    assert entry.unique_id == f"{config_entry.entry_id}_battery.charge"
 
-    state = hass.states.get("sensor.ups1_battery_charge")
-    assert state.state == "100"
-
-    expected_attributes = {
-        "device_class": "battery",
-        "friendly_name": "Ups1 Battery charge",
-        "unit_of_measurement": PERCENTAGE,
-    }
-    # Only test for a subset of attributes in case
-    # HA changes the implementation and a new one appears
-    assert all(
-        state.attributes[key] == attr for key, attr in expected_attributes.items()
-    )
-
-
-async def test_blazer_usb(hass: HomeAssistant) -> None:
-    """Test creation of blazer_usb sensors."""
-
-    config_entry = await async_init_integration(hass, "blazer_usb")
-    registry = er.async_get(hass)
-    entry = registry.async_get("sensor.ups1_battery_charge")
-    assert entry
-    assert entry.unique_id == f"{config_entry.entry_id}_battery.charge"
+    if unique_id:
+        assert entry.unique_id == unique_id
+    else:
+        assert entry.unique_id == f"{config_entry.entry_id}_battery.charge"
```

Comment:
Ideally we want to avoid branching in tests.

I'd make one test for ones with unique ids and ones without

Reply:
Ok, didn’t know that, will do 👍

Reply:
Done